File: src/main/java/carelog/carelog/user/web/dto/UserResponse.java
package carelog.carelog.user.web.dto;

import carelog.carelog.user.domain.ManagerType;
import carelog.carelog.user.domain.User;
import carelog.carelog.user.domain.UserRole;
import io.swagger.v3.oas.annotations.media.Schema;

import java.util.UUID;

public record UserResponse(
        @Schema(description = "사용자 공개 ID") UUID publicId,
        @Schema(description = "사용자 로그인 ID") String userId,
        @Schema(description = "이메일") String email,
        @Schema(description = "이름") String name,
        @Schema(description = "역할") UserRole role,
        @Schema(description = "직군 (매니저만 해당)") ManagerType managerType,
        @Schema(description = "암호화된 전화번호") String phoneEncrypted,
        @Schema(description = "암호화된 주소") String addressEncrypted
) {
    public static UserResponse from(User user) {
        return new UserResponse(
                user.getPublicId(),
                user.getUserId(),
                user.getEmail(),
                user.getName(),
                user.getRole(),
                user.getManagerType(),
                user.getPhoneEncrypted(),
                user.getAddressEncrypted()
        );
    }
}

File: src/main/java/carelog/carelog/user/web/dto/UserResponse.kt
package carelog.carelog.user.web.dto

import carelog.carelog.user.domain.User
import carelog.carelog.user.domain.UserRole

data class UserResponse (
    val id: Long,
    val userId: String,
    val email: String,
    val name: String?,
    val role: UserRole,
    val phoneEncrypted: String?,
    val addressEncrypted: String?
){
    companion object{
        fun from(user: User): UserResponse = UserResponse(
            id = user.id,
            userId = user.userId,
            email = user.email,
            name = user.name,
            role = user.role,
            phoneEncrypted = user.phoneEncrypted,
            addressEncrypted = user.addressEncrypted
        )
    }
}
